src/libcore/array.rs

```diff
@@ -406,6 +408,73 @@ mod impls_using_const_generics {
         }
     }
 
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    impl<T, const N: usize> IntoIterator for [T; N]
+    where
+        [T; N]: LengthAtMost32,
+    {
+        type Item = T;
+        type IntoIter = IntoIter<T, {N}>;
+
+        fn into_iter(self) -> Self::IntoIter {
+            IntoIter {
+                data: ManuallyDrop::new(self),
+                pos: 0,
+            }
+        }
+    }
+
+    /// A by-value array iterator.
+    ///
+    /// This struct is created by the `IntoIterator::into_iter`] method on
+    /// arrays.
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    pub struct IntoIter<T, const N: usize> {
+        /// This is the array we are iterating over. Elements that were already
+        /// yielded (the ones with index `< self.pos`) have been read via
+        /// `ptr::read` and must not be accessed anymore!
+        data: ManuallyDrop<[T; N]>,
+        pos: usize,
+    }
+
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    impl<T, const N: usize> Iterator for IntoIter<T, {N}> {
```

Comment:
nit: the `impl` is insta-stable, so consider marking stable in 1.38.  (Or 1.39 if this takes more than 3 weeks.)

Reply:
I understand that impls are not checked for stability by the compiler, but it seems a bit strange to mark them as `stable` IMO. Furthermore, to mark them as stable, one has to use another feature name as one feature cannot be simultaneously stable and unstable.

Should I still change it or is it fine to keep the impls as unstable for now?

Reply:
> I understand that impls are not checked for stability by the compiler, but it seems a bit strange to mark them as `stable` IMO.

Not checking for stability does render them stable so using `#[stable(...)]` better reflects reality whereas `#[unstable(...)]` would be misleading.

> Furthermore, to mark them as stable, one has to use another feature name as one feature cannot be simultaneously stable and unstable.

New feature gate names are as cheap as making up a new string... it's totally fine. :)

> Should I still change it or is it fine to keep the impls as unstable for now?

Yep, change it.